src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb

```diff
@@ -1698,6 +1698,501 @@ End Class
+first
+3
+second
+5
+            ]]>)
+        End Sub
+
+        <Fact()>
+        Public Sub Inference07()
+
+            Dim verifier = CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+Imports System
+Module Module1
+    Sub Main()
+        Test(Function(x)(x, x), Function(t)1)
+        Test1(Function(x)(x, x), Function(t)1)
+        Test2((a:= 1, b:= 2), Function(t)(t.a, t.b))
+    End Sub
+
+    Sub Test(Of U)(f1 as Func(of Integer, ValueTuple(Of U, U)), f2 as Func(Of ValueTuple(Of U, U), Integer))
+        System.Console.WriteLine(f2(f1(1)))
+    End Sub
+
+    Sub Test1(of U)(f1 As Func(of integer, (U, U)), f2 as Func(Of (U, U), integer))
+        System.Console.WriteLine(f2(f1(1)))
+    End Sub 
+
+    Sub Test2(of U, T)(f1 as U , f2 As Func(Of U, (x as T, y As T)))
+        System.Console.WriteLine(f2(f1).y)
+    End Sub
+End Module
+
+    </file>
+</compilation>, additionalRefs:={ValueTupleRef, SystemRuntimeFacadeRef}, expectedOutput:=<![CDATA[
+1
+1
+2
+            ]]>)
+        End Sub
+
+        <Fact()>
+        Public Sub InferenceChain001()
+
+            Dim verifier = CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+Imports System
+Module Module1
+    Sub Main()
+        Test(Function(x As (Integer, Integer)) (x, x), Function(t) (t, t))
+    End Sub
+
+    Sub Test(Of T, U, V)(f1 As Func(Of (T,T), (U,U)), f2 As Func(Of (U,U), (V,V)))
+        System.Console.WriteLine(f2(f1(Nothing)))
+
+        System.Console.WriteLine(GetType(T))
+        System.Console.WriteLine(GetType(U))
+        System.Console.WriteLine(GetType(V))
+    End Sub
+
+
+End Module
+
+    </file>
+</compilation>, additionalRefs:={ValueTupleRef, SystemRuntimeFacadeRef}, expectedOutput:=<![CDATA[
+(((0, 0), (0, 0)), ((0, 0), (0, 0)))
+System.Int32
+System.ValueTuple`2[System.Int32,System.Int32]
+System.ValueTuple`2[System.ValueTuple`2[System.Int32,System.Int32],System.ValueTuple`2[System.Int32,System.Int32]]
+
+            ]]>)
+        End Sub
+
+        <Fact()>
+        Public Sub InferenceChain002()
+
+            Dim verifier = CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+
+Imports System
+Module Module1
+    Sub Main()
+        Test(Function(x As (Integer, Object)) (x, x), Function(t) (t, t))
+    End Sub
+
+    Sub Test(Of T, U, V)(ByRef f1 As Func(Of (T, T), (U, U)), ByRef f2 As Func(Of (U, U), (V, V)))
+        System.Console.WriteLine(f2(f1(Nothing)))
+
+        System.Console.WriteLine(GetType(T))
+        System.Console.WriteLine(GetType(U))
+        System.Console.WriteLine(GetType(V))
+    End Sub
+
+
+End Module
+
+    </file>
+</compilation>, additionalRefs:={ValueTupleRef, SystemRuntimeFacadeRef}, expectedOutput:=<![CDATA[
+(((0, ), (0, )), ((0, ), (0, )))
+System.Object
+System.ValueTuple`2[System.Int32,System.Object]
+System.ValueTuple`2[System.ValueTuple`2[System.Int32,System.Object],System.ValueTuple`2[System.Int32,System.Object]]
+
+            ]]>)
+        End Sub
```

Comment:
It's not directly for the PR, but I didn't find a test using tuple syntax but without referencing the ValueTuple library.
In C#, it reports ERR_PredefinedTypeMemberNotFoundInAssembly.